federation/pkg/federation-controller/service/endpoint_helper.go

```diff
@@ -31,22 +31,42 @@ import (
 // worker runs a worker thread that just dequeues items, processes them, and marks them done.
 // It enforces that the syncHandler is never invoked concurrently with the same key.
 func (sc *ServiceController) clusterEndpointWorker() {
-	fedClient := sc.federationClient
+	// process all pending events in endpointWorkerDoneChan
+	eventPending := true
```

Comment:
Oops, you are right. This isn't a common idiom which is why it threw me off the guard. 

What you have is fine, but if you want to stay with a commonly used idiom, please see https://golang.org/ref/spec#Break_statements.